index.js

```diff
@@ -369,17 +369,24 @@ class ResembleHelper extends Helper {
     const helper = this._getHelper();
     await helper.waitForVisible(selector);
     const els = await helper._locate(selector);
-    if (!els.length) throw new Error(`Element ${selector} couldn't be located`);
-    const el = els[0];
+    
+    if (this.helpers['TestCafe']) {
+      if (await els.count != 1) throw new Error(`Element ${selector} couldn't be located or isn't unique on the page`);
+    }
+    else {
+      if (!els.length) throw new Error(`Element ${selector} couldn't be located`);
+    }
 
     let location, size;
 
     if (this.helpers['Puppeteer']) {
+      const el = els[0];
       const box = await el.boundingBox();
       size = location = box;
     }
 
     if (this.helpers['WebDriver'] || this.helpers['Appium']) {
+      const el = els[0];
```

Comment:
also here?

Reply:
yes for the same reason. I didn't like this duplicate of code. But If you have a better solution, I'm ready to implement it.